users/api.py

```diff
@@ -21,58 +27,123 @@ def get_user_by_id(user_id):
     return User.objects.get(id=user_id)
 
 
-def fetch_user(user_property):
+def _is_case_insensitive_searchable(field_name):
+    """
+    Indicates whether or not a given field in the User model is a string field
+
+    Return:
+         bool: True if the given User field is a string field
+    """
+    return field_name in CASE_INSENSITIVE_SEARCHABLE_FIELDS
+
+
+def _determine_filter_field(user_property):
+    """
+    Assesses the value provided to search for a User and returns the field name that should
+    be used for the query (e.g.: "id", "username", "email")
+
+    Returns:
+         str: User field name that should be used for a query based on the value provided
+    """
+    if isinstance(user_property, int) or user_property.isdigit():
+        return "id"
+    else:
+        try:
+            validate_email(user_property)
+            return "email"
+        except ValidationError:
+            return "username"
+
+
+def fetch_user(user_property, ignore_case=True):
     """
     Attempts to fetch a user based on several properties
 
     Args:
         user_property (Union[str, int]): The id, email, or username of some User
+        ignore_case (bool): If True, the User query will be case-insensitive
     Returns:
         User: A user that matches the given property
     """
-    if isinstance(user_property, int):
-        return User.objects.get(id=user_property)
-    elif user_property.isdigit():
-        return User.objects.get(id=int(user_property))
+    filter_field = _determine_filter_field(user_property)
+    if filter_field == "id" and isinstance(user_property, str):
```

Comment:
I don't think this is necessary here, Django will handle both `int` and `str` values for filter values.

Reply:
TIL! I'll change